### PR TITLE
Remove macros

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description

This PR removes macros! 🥳 

### Testing this PR

Nothing should really change at API level. At all. The Management API does not expose nor concern itself with macros.

Test this by:

1. Verify that the `cmsMacro` and `cmsMacroProperty` tables are removed from the database once the site starts (on an existing V13/V14 database).
2. Get a partial view snippet - this should _not_ fail.
3. Smoke test the Management API.
4. Run the installer (or install unattended) and verify that the `cmsMacro` and `cmsMacroProperty` tables are not created at install time.

### CODE REVIEW

File: src/Umbraco.Core/Deploy/Direction.cs [hunk 1]
```diff
@@ -1,16 +0,0 @@
-namespace Umbraco.Cms.Core.Deploy;
```
Comment (bergmania) [inline src/Umbraco.Core/Deploy/Direction.cs:1]: do the DXP know about this?

File: src/Umbraco.Core/Services/IFileService.cs [hunk 2]
```diff
@@ -8,19 +8,15 @@ namespace Umbraco.Cms.Core.Services;
 /// </summary>
 public interface IFileService : IService
 {
-    [Obsolete("Please use SnippetCollection.GetPartialViewSnippetNames() or SnippetCollection.GetPartialViewMacroSnippetNames() instead. Scheduled for removal in V12.")]
+    [Obsolete("Please use SnippetCollection.GetPartialViewSnippetNames() instead. Scheduled for removal in V12.")]
```
Comment (bergmania) [inline src/Umbraco.Core/Services/IFileService.cs:11]: The method seems safe to remove now also.

**Review by bergmania — APPROVED** [review]
If the change with the deploy file is okay with the DXP group, you can merge.. Tests out good

**Comment by KevinJump** [comment]
this should probably be in announcements  ? https://github.com/umbraco/Announcements/issues

**Comment by onlinegroeicoach** [comment]
Congratulations you just killed Umbraco for a ton of users. 

Think Umbraco forgets that a lot of the users have been using Umbraco for a long time, long before things like Block Editors were a thing. So we relied on things like Macro's and other things which are no longer supported in the newest version. So think a lot of people with big websites with hunderts of pages will be stuck on older versions. Not to mention cloud users who are forced to upgrade.

**Comment by wicketbr** [comment]
Unfortunately, this will spell the end of Umbraco for us. Of the over 9000 content pages in our Umbraco system, over 2000 pages use at least one macro and many more pages are involved in macros, such as macros that roll content up into accordions, tabbed dialogs, contact thumbnails, calendars, content rotators, and FAQ pages. In addition, we chose Umbraco because our developers could add custom solutions using .NET in macros. We pull data from databases in macros and build out dynamic content from those external data sources, we embed forms into page from our in-house forms system, and so much more. We will be stuck now on version 13 forever, which eventually will become a security concern as there will come a time when it will no longer receive updates. Macros were one of the main reasons we chose Umbraco so many years ago as it gave us the ability to easily allow our end users to do complex things that a CMS can't do natively. I just happened to be looking at what was coming in version 14 and saw this tiny note about macros being removed. I am sure so many more customers depend on macros and aren't even aware they are going away. Without macros Umbraco will just be a boring, static, website builder.

**Comment by wicketbr** [comment]
Is there any replacement for macros in this new version. We cannot move to Umbraco 14 without a similar feature. I installed the new version and didn't see anything that could be inserted in the rich editor like partial views, etc., that would replace macros. I've been using Umbraco since 2012 and I would hate to have to switch over to another CMS to get the support of macros. We do so much custom coding in Umbraco to support our customers, not to mention that we have macros that do things like rollup child content onto a page. I'm not against moving to some other technology, like partial views, as long as they can be inserted in the rich editor with parameters.

**Pull request merged**